### ISSUE TITLE
Articulos Publicar Feb 13

### ISSUE THREAD

**Issue by Jsolis4geeks** [4GeeksAcademy/About-4Geeks-Academy#4374]
Miami: 3
Latam: 2

**Comment by Jsolis4geeks** [comment]
https://admin.4geeks.com/media/asset/how-to-square-a-number-in-python#

---
title: "How to square a number in python"
subtitle: "On how to square a number in Python we explore different ways that includes pow(), math.pow, the power operator (**) as we create as well our own function to receive exactly this squared value"
tags: ["python", "power", "square"]
date: "2023-02-06T16:36:30+00:00"
authors: ['Javier Seiglie']
status: "draft"

---

```python
square = pow(6, 2)
print(square)
#Output-> 36
```

Whether you're a beginner or an experienced programmer, this article provides a comprehensive guide to squaring numbers in [Python](https://4geeks.com/lesson/intro-to-python). The step-by-step instructions and clear explanations make it easy to understand and apply the concepts. So if you're interested in learning how to square numbers in Python, be sure to give this article a read!

## Creating the function ourself:

The first method we are presenting is exactly, creating a function that will return the squared number that receive. 

As you should already know, a squared number is the number multiplyed by itself, so, let say the square of `2` would be `4` because `2*2=4`. If we were to choose `5`, we would receive `25` and so on.

So, keeping that in mind, we can create our function:

```python
def square(num):
    return num*num

print(square(3))
#Output-> 9
```

## Using the power(square) operator:

Python comes with a built in with a power (square) operator `**`. The syntax is quite straight forward as all the other logic or math operators.

```python
def squareOperator(base):
    return base ** 2

print(squareOperator(4))
#Output-> 16
```

## Using the power() function:

As the power operator, the power() function comes built-in and ready to work on Python. The syntax is as simple as:

```python
def squarePow(base, power):
    return base ** power

print(squarePow(5, 2))
#Output-> 25
```

## Using the math.pow():

The only difference between the `math.pow()` and `pow` resides in the first needs the math module imported returns always a float number, even if the result of the operation has not a floating point (if you want to learn more about Float numbers, check [What is a float in Python](https://4geeks.com/how-to/what-is-float-in-python "What is a float in Python")).

```python
import math

def squareMathPow(base, power):
    return math.pow(base, power)

print(squareMathPow(5,2))
#Output-> 25
```

You can find more of these articles at [4Geeks](https://4geeks.com/). Hope you enjoyed the reading and keep on the Geek side!

**Comment by Jsolis4geeks** [comment]
https://admin.4geeks.com/media/asset/how-to-exit-python-in-terminal#

---
title: "How to exit Python in terminal"
subtitle: "Exiting Python from the terminal can be achieved in 3 different ways. The exit and quit commands will exit python and you can as well exit by using keyboard shortcuts as ctrl+z and ctrl+d on Windows and Unix Based Systems respectively"
tags: ["python", "exit"]
date: "2023-02-06T16:36:30+00:00"
authors: ['Javier Seiglie']
status: "draft"

---

From using the quit() and exit() functions to keyboard shortcuts, this article covers everything you need to know to exit [Python](https://4geeks.com/lesson/intro-to-python) in the terminal. So if you're looking for a comprehensive guide on exiting Python in the terminal, be sure to give this article a read!

```python
exit()
```

## Using Exit()

To exit Python on terminal, can be achieved in 3 ways:

### Exit()

The `exit()`command terminates Python in a safe way and stores your data correctly. It's important to write the `()` after the word, or Python could understand that you want to know what `exit` is and return that is a function and ask for you to confirm. 

```python
Use exit() or Ctrl-Z plus Return to exit
```

This is the reason why we're emphasizing the presence of the `()` at the end.

The syntax is quite straight-forward:

```python
exit()
```

### Using quit()

The command `quit()` follows the same syntax as `exit()`. You'll need to add the `()` at the end or you'll receive the same message as before.

```python
Use quit() or Ctrl-Z plus Return to exit
```
### Using shortcuts

Keyboard shortcuts are a common use among us developers because they save time and make you work faster. Here you have the shortcuts for Windows and Unix Systems (Linux and MacOs).

#### Using ctrl+z on Windows

As stated before, on the message that you receive when passing `exit` without the `()`, `control + z` will exit Python from the terminal as well. 

Before pressing `ctrl+z` to exit Python just make sure your Python terminal is the selected windows, and then press `ctrl`, hold it down and press `z` and then `Return`.

#### Using ctrl+d on MacOs and Linux

On Unix systems such as MacOs or Linux, `ctrl+z` will not exit Python on the terminal. 

The keyboard shortcut for exiting Python on this systems would be `ctrl + d` and on these systems you won't need to confirm with `return` because their terminal will receive and execute this command.

We will do as we did with the Windows shortcut. Press and hold `ctrl` and then press `d` to exit Python from the terminal making sure the terminal is the selected window.

You can read more related topics at [4Geeks](https://4geeks.com/). Hope you enjoy the reading and keep on the Geek side!

**Comment by Jsolis4geeks** [comment]
https://admin.4geeks.com/media/asset/how-to-concatenate-strings-on-python#

---
title: "How to concatenate Strings on Python"
subtitle: "On how to concatenate Strings in Python we explore different ways to concatenate and your choosing will depend on your style and needs."
tags: ["python"]

---

```python 
str1 = "Hello"
str2 = "Geeks"
int1 = 4
print(f"{str1} from {int1} {str2} Academy!")
#Output -> Hello from 4 Geeks Academy!
```

String concatenation is a common action we, as [software developers](https://4geeksacademy.com/us/coding-bootcamps/part-time-full-stack-developer), have to deal with in almost any project. This is the reason why we can achieve this `union of texts` in different ways, depending on style and the need of the data we are presenting in our project.

## Concatenate using the + operator

Making use of the `+` operator is the classic way to concatenate strings. You "add" the variable to your text but remember always to make use of spaces for the string rendering to be as you want.

Implementation: 

```python
#Concatenate using the + operator
str1 = "Geeks"
print("Showing the str1 variable: " + str1 +  " using the + operator")
#Output-> Showing the str1 variable: Geeks using the + operator
```

Take note that there's a space between "variable: " and " using". If we wouldn't place that space, the output would look like this:

```python
#Output-> Showing the str1 variable:Geeksusing the + operator
```
The + operator will "add" the variable to the text and will not add spaces.

## Concatenate using the += Operator

We can use the += Operator to concatenate strings as well, it's pretty similar to using the `+` Operator, but different since we are adding text to a variable.

```python
#Concatenate using the += operator
text = "Python"
str1 = "<3"
text+= str1
print(text)
#Output-> Python<3
```

Since we are "adding" string values to our variable **text**, these values will be concatenated as they are, no space will be automatically added.

### Concatenate a int or foat value with text with str()

Numbers and strings are data type completely different. When we want to add a number (int, float, etc) to a string, we will need to use the `str()` to convert that number.

```python
#Concatenate using the + Operator with numbers
str1 = "Geeks"
int1 = 4
print("At " + str(int1) + " " + str1 +  " Academy we love Python! <3")
#Output-> At 4 Geeks Academy we love python! <3
```

As stated above, we are adding spaces for the render to respect writing rules (all words and numbers separated from one another).

If we would not convert the `int1` number variable to a **string**, we would receive the following error letting you know you are trying to concatenate an Integer with a String.

```python
TypeError: can only concatenate str (not "int") to str
```


## Concatenate using the f-Strings

Python f-Strings (formatted string literals) makes writing strings in a more clear and understanding way for humans. This option was [introduced on Python](https://4geeks.com/lesson/intro-to-python) 3.6 so keep in mind backward compatibility when using it.

This is one of the most used methods among experienced developers because of the ease to insert `variables` into the **square brackets**. The variable(s) will be evaluated and a string representation will be displayed. 

Implementation: 

```python
#Concatenate using the f-Strings method
str1 = "Geeks"
print(f"Showing the str1 variable: {str1} using f-String")
#Output-> Showing the str1 variable: Geeks using f-String
```

Making use of the `f-Strings` concatenation, we don't need to convert numbers (Int, Float, etc) to string to be able to display them inside the curly brackets.

```python
#Concatenate using the f-Strings method with numbers 
str1 = "Geeks"
int1 = 4
print(f"At {int1} {str1} Academy we love Python! <3")
#Output-> At 4 Geeks Academy we love python! <3
```

## Concatenate using .format()

Another way to concatenate in Python is by making use of the `format()` method. It works similar to `f-Strings` since they both use curly brackets `{}` to insert variables but is available from version 2.7.

This method doesn't need for the number values to be converted into strings.

```python
#Concatenate using the .format() method
str1 = "Geeks"
str2 = "Academy"
int1 = 4
text = "{}{}{}".format(int1, str1, str2)
print(text)
#Output-> 4GeeksAcademy
```

As you can see, the result returned is the expected, but with spacing between words.

## Concatenate using %

The `%` operator is an older way to concatenate strings that work very similar to the '.format()` method. We use `%s` as placeholders for different values, as we did with the curly brackets in previous examples.

```python
#Concatenate using the % Operator
str1 = "Geeks"
str2 = "Academy"
int1 = 4
text = "%s%s%s" % (int1, str1, str2)
print(text)
#Output-> 4GeeksAcademy
```

## Concatenate using the * Operator 

The use of the `*` Operator is not a concatenation in the way we have been doing so far. The `*` Operator is used for multiplication and that's exactly what it does, multiplying for the amount given the string.

```python
#Concatenate using the * Operator
str1 = "4 Geeks Academy "
print(str1*3)
#Output-> 4 Geeks Academy 4 Geeks Academy 4 Geeks Academy 
```

If you format the **string** variable correctly (leaving a space at the end of the sentence/word), this is the best way to repeat a line of text as many times as you may need.


You can read more related topics at [4Geeks](https://4geeks.com/lesson/intro-to-python). Hope you enjoyed the reading and keep on the Geek side!

**Comment by Jsolis4geeks** [comment]
https://admin.4geeks.com/media/asset/coding-bootcamp-in-berlin#

---
author: "jrdelrio"
excerpt: ""Transform your career with 4Geeks Academy's Coding Bootcamp in Berlin. Enhance your skills and join the tech industry.""
title: "Coding Bootcamp in Berlin"
template: "post"
---

## Definition of Coding Bootcamp

Generally, to have all the knowledge to be a programmer, you must enter a university and study the entire degree and then go to work. But, with the technological growth of recent times, the demand for developers has grown exponentially to become one of the most demanded professions in the entire world. A [coding bootcamp](https://4geeksacademy.com/) is a short but highly demanding intensive course that prepares anyone with no previous knowledge of programming or development, to a person ready to start their career in technology and enter the job market.

There are bootcamps of different themes, prices, modalities, etc… But essentially they are all the same goals. 4Geeks Academy, for example, offers 2 different bootcamps for beginners, in three different disciplines: [**full stack development**](https://4geeksacademy.com/us/coding-bootcamps/part-time-full-stack-developer) and [**data science and machine learning**](https://4geeksacademy.com/us/coding-bootcamps/datascience-machine-learning) . Any of these, in 6 months prepares you to go out into the world of work in addition to offering you help, after graduation, to position yourself in your first job.


## Overview of Coding Bootcamp in Berlin

There are a huge number of bootcamps in Germany and especially in Berlin, because this city is a source of tech start-ups and with a huge technological growth. Almost every tech academy that has bootcamps in Europe has a branch office in Berlin. Also Berlin is one of the most searched city for jobs in the area of technology. As a Top institution 4Geeks Academy has a [Coding Bootcamp in Berlin](https://4geeksacademy.com/us/coding-campus/coding-bootcamp-berlin-germany) as long as in other major cities like the [Coding Bootcamp from Hamburg](https://4geeksacademy.com/us/coding-campus/coding-bootcamp-hamburg-germany) and other [Bootcamp in Munich](https://4geeksacademy.com/us/coding-campus/coding-bootcamp-munich-germany).

### Benefits of Attending a Coding Bootcamp in Berlin

The vast majority of bootcamps that can be found on the internet have full-time, part-time, face-to-face and remote modalities. Each modality has its advantages and disadvantages, but without a doubt, attending a bootcamp in Berlin face-to-face has special benefits, such as.

+ **Networking**: Immersing yourself in the world of people dedicated to technology will bring into your life a **large number of contacts** that will undoubtedly help your professional growth once you start working. Bootcamps provide opportunities to **network** with other aspiring and established tech professionals. Teachers, classmates, orators, and guests serve to meet people and build a network of contacts from the time you are a student and beyond.

+ **Career Opportunities**: As mentioned above, Berlin is a hub for tech startups and established tech companies, one of the biggest in the world. Attending a bootcamp in Berlin can increase your chances of finding a job in the tech industry, the job demand there is enormous and there are avant-garde companies interested in making something different, something that developers love.


### Career Advancement Opportunities

Given the sheer size of the technology industry in this city, the opportunities for growth are enormous. As a developer, you can scale in experience like any employee of a technology company, starting with Junior, semi-senior and then Senior, reaching great salaries and job positions. Then the chief technology officer (CTO) of a large company also submits a job position for more experienced persons, and so on.

## Popular Coding Bootcamps in Berlin

So, being Berlin a city with many technology companies, start-ups, bootcamps and opportunities, the most prestigious and popular bootcamps in this city are:

+ **4Geeks Academy**. It offers 2 bootcamp options in more than 20 countries, including the United States, Europe, South America, and Canada.

+ **Le Wagon**. Offer Web Development and Data Science bootcamp at over 40 campuses across the world.

+ **IronHack**. Offers full-time and part-time bootcamps in Web Development, UX/UI design, Data Analytics and Cyber Security.

+ **Spiced Academy**. Offers full-time, 12-week coding bootcamps and data science bootcamps in Berlin.


## Preparing for a Coding Bootcamp in Berlin

Many academies around the world give to their students when enrolling a bunch of exercises to give a basic understanding of the tools that they are about to learn. Some of them are reading lessons and others are logic exercises. For example,[ 4Geeks Academy](https://4geeksacademy.com/), prepares you with more than **150 javaScript exercises** and a lot of reading lessons about how the internet works, and exercises of HTML and CSS, so when you start everybody is familiarized with the basic concepts of web development.


### Prerequisites for Enrolling

One of the things that make boot-camps such a great opportunity for everyone is that **no previous knowledge** of any of the tools that will be used or any knowledge of technology is required. The teachers and the educational line are in charge of training you as a developer from scratch. They start with the most basic concepts, and with the exercises they propose, which are arranged in such a way that the level of difficulty increases in a pedagogical way, and they make sure that once the course has started you will have no problems implementing the tools they teach. Even, at 4Geeks Academy there is a mentoring system, which consists of a scheduled meeting with a teacher or former student of the academy who can guide you if you find yourself stuck due to any difficulty or lack of knowledge or understanding of a subject and this works forever, **even after boot-camp is over**.


### Recommended Skills to Develop

When a person [becomes a developer](https://4geeksacademy.com/us/landing/learn-to-code), the least part of their time is spent writing code, and most of their time is spent reading code and looking for information. The ability to **searching information in Google** for the necessary solutions for a certain task is essential for the training of any developer, understanding that generally someone has already gone through that problem in the past and has remained in a blog to help those of the future. The most used blogs to search for information are **stackoverflow.com** and **developer.mozilla.org**.

Another skill that needs to be developed to be successful as a developer is **resilience**, as there will be many hours of trying different methods to achieve a result that often will not work. Codes are extremely perfectionist and a misspelled letter can break the code and bring down a tower of commands even if it is very well built.

As you can imagine, the **English language** is something that you have to practice a lot if you aren’t very skilled. The industry and all the information of technology is dominated by the English language.


### Tips for Succeeding in a Coding Bootcamp

There are different ways to approach the learning process of a bootcamp, and they are all correct. But, without a doubt, there are some tips that are transversal for any student who is starting a bootcamp:

+ **Study the lesson before class**. 
Coming to class without having had a minimal approach to the topic that will be discussed in class will cause many conceptual doubts to arise during the class. Assuming also that there are more than 10 people present in the class, these doubts can be many and the class can have a much lower performance than it would be if all the students read the lesson and then only go with the doubts to the class. In this way, the whole course and each student separately will have a better use of time and a much better and more efficient learning performance.

+ **Review at least every day**. 
Coding is like a high-performance sport, that is, you have to be trained. It is surprising the speed with which certain learning is forgotten when it is not practiced and is not constantly trained to continue development. For this reason, it is always highly recommended to spend at least one hour a day reviewing the material, looking at the code, doing exercises, or whatever keeps you fit to face future problems.

+ **Do not leave behind deliveries or matters not understood**. 
A bootcamp has the characteristic of being very demanding in terms of the speed of learning. In 6 months, or even less, a person who knows nothing about the subject is prepared from scratch. For this reason, if the rhythm of assignments and projects is neglected, these can accumulate and can be very stressful for the student, not only because they accumulate quickly, but also, as mentioned in the previous point, they go from one subject to another very quickly, so tackling an old subject can require a lot more work than doing it on time. Organizing personal times is crucial to be able to carry out learning at the required pace, schedule the hours of the day necessary to avoid falling behind and to read the classes before the chair.


## Salaries for Coding Bootcamp graduates in Berlin

The average [salary for bootcamp graduates in Berlin](https://4geeksacademy.com/us/software-engineer-salary/software-engineer-salary-germany) depends on the specific program, industry area, and the individual's level. However, a full-stack developer in Berlin, just graduate can expect to earn around **€60,000 - €80,000** per year and senior positions offering salaries upwards of **€100,000**.


## Job Opportunities in Berlin's Tech Industry

Berlin is one of the cities with the most job opportunities in technology companies in the world, with good salaries, good contracts, and a very good work network. Not only are there thousands of startups launching every year, but all the big companies have offices with jobs in this city. These include [Microsoft](https://www.microsoft.com/en-us/), GitHub, Airbnb, SAP, Wayfair, Amazon, Datadog, and Snowflake. Many of these offer 100% remote jobs and other face-to-face and hybrid modalities.
